packages/iTwinUI-react/src/core/DatePicker/DatePicker.test.tsx

```diff
@@ -307,3 +307,102 @@ it('should navigate between years', () => {
+      onChange={onClick}
+      isRange={true}
+    />,
+  );
+  assertMonthYear(container, 'June', '2020');
+  let selectedStartDay = container.querySelector(
+    '.iui-calendar-day-range-start',
+  ) as HTMLElement;
+  let selectedRange = container.querySelectorAll('.iui-calendar-day-range');
+  let selectedEndDay = container.querySelector(
+    '.iui-calendar-day-range-end',
+  ) as HTMLElement;
+  expect(selectedStartDay.textContent).toBe('5');
+  expect(selectedRange).toHaveLength(4);
+  expect(selectedEndDay.textContent).toBe('10');
+
+  const startDay = getByText('15');
+  const endDay = getByText('20');
+  fireEvent.click(startDay);
+  fireEvent.click(endDay);
+
+  selectedStartDay = container.querySelector(
+    '.iui-calendar-day-range-start',
+  ) as HTMLElement;
+  selectedRange = container.querySelectorAll('.iui-calendar-day-range');
+  selectedEndDay = container.querySelector(
+    '.iui-calendar-day-range-end',
+  ) as HTMLElement;
+  expect(selectedStartDay.textContent).toBe('15');
+  expect(selectedRange).toHaveLength(4);
+  expect(selectedEndDay.textContent).toBe('20');
+});
+
+it('should update startDate when selecting an endDate value that is before startDate', () => {
+  const onClick = jest.fn();
+  const { container, getByText } = render(
+    <DatePicker
+      startDate={new Date(2020, 5, 5)}
+      endDate={new Date(2020, 5, 20)}
+      onChange={onClick}
+      isRange={true}
+    />,
+  );
+  assertMonthYear(container, 'June', '2020');
+
+  const startDay = getByText('15');
+  const endDay = getByText('12');
+  fireEvent.click(startDay);
+  fireEvent.click(endDay);
+
+  const selectedStartDay = container.querySelector(
+    '.iui-calendar-day-range-start',
+  ) as HTMLElement;
+  const selectedRange = container.querySelectorAll('.iui-calendar-day-range');
+  const selectedEndDay = container.querySelector(
+    '.iui-calendar-day-range-end',
+  ) as HTMLElement;
+  expect(selectedStartDay.textContent).toBe('12');
+  expect(selectedRange).toHaveLength(7);
+  expect(selectedEndDay.textContent).toBe('20');
+});
+
+it('should update endDate when selecting a startDate value that is after endDate', () => {
+  const onClick = jest.fn();
+  const { container, getByText } = render(
+    <DatePicker
+      startDate={new Date(2020, 5, 5)}
+      endDate={new Date(2020, 5, 10)}
+      onChange={onClick}
+      isRange={true}
+    />,
+  );
+  assertMonthYear(container, 'June', '2020');
+
+  const startDay = getByText('15');
+  fireEvent.click(startDay);
+
+  const selectedDay = container.querySelector(
+    selectedDaySelector,
+  ) as HTMLElement;
+  const selectedStartDay = container.querySelector(
+    '.iui-calendar-day-range-start',
+  ) as HTMLElement;
+  const selectedRange = container.querySelectorAll('.iui-calendar-day-range');
+  const selectedEndDay = container.querySelector(
+    '.iui-calendar-day-range-end',
+  ) as HTMLElement;
+  expect(selectedDay.textContent).toBe('15');
+  expect(selectedStartDay).toBeNull();
```

Comment:
Why it is null and range length 0?

Reply:
Hm, I checked the other pickers and when you already have start and end day set and you click after end day, they select it as start day and there is hover state showing that you need to select end day:
![date-picker-range-past-end-day](https://user-images.githubusercontent.com/36186912/189053010-4f50e150-a487-41f7-bb76-6e5504f0e497.gif)

Reply:
Unless we use the same highlighting CSS for hover as we do for selected range, this would require more CSS changes, no?